pkg/virt-api/rest/authorizer.go

```diff
@@ -43,6 +42,34 @@ const (
 	userExtraHeaderPrefix = "X-Remote-Extra-"
 )
 
+var noAuthEndpoints = map[string]struct{}{
+	"/":           {},
+	"/healthz":    {},
+	"/openapi/v2": {},
+	"/apis/subresources.kubevirt.io/v1/version":       {},
+	"/apis/subresources.kubevirt.io/v1/guestfs":       {},
+	"/apis/subresources.kubevirt.io/v1/healthz":       {},
+	"/apis/subresources.kubevirt.io/v1alpha3/version": {},
+	"/apis/subresources.kubevirt.io/v1alpha3/guestfs": {},
+	"/apis/subresources.kubevirt.io/v1alpha3/healthz": {},
+	// FIXME Do we really want to allow unauthenticated access to these endpoints?
+	"/apis":                                   {},
+	"/apis/subresources.kubevirt.io":          {},
```

Comment:
The apiserver was at least in the past accessing these paths as part of an aggregated apiserver service discovery. It did that without being authenticated. We can try removing some, if they are not needed, but we have to carefully check the apiserver logs, because we can easily  crash the auto-dicovery without clearly seeing it (e.g. ours still gets registered, but others can't register anymore afterwards ,... ). For as long as the apiserver of all relevant k8s versions does not  throw any errors we are good.

Reply:
I was finally able to test it and the result was that access to the versioned endpoints is required for the apiserver discovery to work. It can be tested with e.g. `kubectl get --raw /apis/subresources.kubevirt.io/v1`. If the permissions are missing this command will fail.